kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/base/OperationSupport.java

```diff
@@ -611,4 +649,12 @@ public static KubernetesClientException requestException(Request request, Except
   public Config getConfig() {
     return config;
   }
+
+  private MediaType getMediaTypeFromBodyContent(String requestBodyContent) {
```

Comment:
Relying on the content of the body to guess the expected kind of patch seems fragile. And the caller of the function should already know the kind of patch provided.

The `kubectl` command line is expecting the user to provide the kind of patch provided. By default it is the 'strategic', whaever the content of the patch. And if the specified type of patch doesn't correspond to the actual content of the patch, we simply get a `Error from server (BadRequest)`.

Probably fabric8 should do the same ? Maybe the `Patchable` interface could expose three new methods, `patch` for json patch, `merge` for json merge patch, `mergeStrategic` for strategic merge patch ?

Reply:
Hmm, good point. @manusa What do you think?